packages/lisk-transactions/src/transactions/base.ts

```diff
@@ -222,199 +227,62 @@ export abstract class BaseTransaction {
 	}
 
 	public validate(): TransactionResponse {
-		const errors = [...this.validateAsset()];
+		const errors = [...this._validateSchema(), ...this.validateAsset()];
 
 		const transactionBytes = this.getBasicBytes();
 
 		const {
-			verified: signatureVerified,
+			valid: signatureValid,
 			error: verificationError,
-		} = verifySignature(
+		} = validateSignature(
 			this.senderPublicKey,
 			this.signature,
 			transactionBytes,
 			this.id,
 		);
 
-		if (!signatureVerified) {
-			errors.push(verificationError as TransactionError);
-		}
-
-		if (Array.isArray(this.signatures) && this.signatures.length > 0) {
-			// Check that signatures are unique
-			const uniqueSignatures: ReadonlyArray<string> = [
-				...new Set(this.signatures),
-			];
-			if (uniqueSignatures.length !== this.signatures.length) {
-				errors.push(
-					new TransactionError(
-						'Encountered duplicate signature in transaction',
-						this.id,
-						'.signatures',
-					),
-				);
-			}
+		if (!signatureValid && verificationError) {
+			errors.push(verificationError);
```

Comment:
Why did we remove the code block for validating signatures array?

Reply:
Uniqueness is validated at schema.
```
signatures: {
			type: 'array',
			uniqueItems: true,
			items: {
				type: 'string',
				format: 'signature',
			},
			minItems: 0,
			maxItems: 15,
		},
```